CSP-1505: Employer account - decline add & permissions request

shaun-downey-education opened the pull request:
- Review and decline add account request shutter page with associated unit testing.

shaun-downey-education committed:
CSP-1505: WIP

shaun-downey-education committed:
Merge branch 'CSP-1504_Review_add_account_and_set_permissions_request' into CSP-1505_Decline_add_and_permission_request

shaun-downey-education committed:
CSP-1505: Decline add account request - Shutter Page.

MarkFCain commented on src/SFA.DAS.Employer.PR.Web/Views/AddAccounts/ReviewAddAccountsRequest.cshtml:
```diff
@@ -61,13 +61,13 @@
                     <div class="govuk-radios" data-module="govuk-radios">
                         <div class="govuk-radios__item">
-                            <input class="govuk-radios__input" id="acceptAddAccountRequestYes" name="AcceptAddAccountRequest" type="radio" value="true">
+                            <input class="govuk-radios__input" id="acceptAddAccountRequestYes" name="AcceptAddAccountRequest" type="radio" value="true" @(Model.AcceptAddAccountRequest.HasValue && Model.AcceptAddAccountRequest.Value ? "checked" : "")>
```
put the 'checked' logic in the viewModel as a string that is empty or "Checked" and add coverage.

MarkFCain commented on src/SFA.DAS.Employer.PR.Web/Views/AddAccounts/ReviewAddAccountsRequest.cshtml:
```diff
@@ -61,13 +61,13 @@
                         </div>
                         <div class="govuk-radios__item">
-                            <input class="govuk-radios__input" id="acceptAddAccountRequestNo" name="AcceptAddAccountRequest" type="radio" value="false">
+                            <input class="govuk-radios__input" id="acceptAddAccountRequestNo" name="AcceptAddAccountRequest" type="radio" value="false" @(Model.AcceptAddAccountRequest.HasValue && !Model.AcceptAddAccountRequest.Value ? "checked" : "")>
```
similar comment as above.

MarkFCain commented on src/SFA.DAS.Employer.PR.Web/Controllers/AddAccountsController.cs:
```diff
@@ -57,34 +57,36 @@ public async Task<IActionResult> Index([FromRoute] Guid requestId, [FromRoute] s
         else
         {
-            // CSP-1505: Decline add & permissions request - Redirect to shutter page for confirmation of decline.
+            return RedirectToRoute(RouteNames.DeclineAddAccount, new { employerAccountId, requestId, acceptAddAccountRequest = acceptRequest });
```
needs coverage

MarkFCain reviewed (commented):
there's a few sonarcloud code smells to address, and some coverage to add

shaun-downey-education committed:
CSP-1505: Additional unit test coverage for AddAccountsController.

Moved radio button logic in the review add account request page into the view model.

shaun-downey-education committed:
CSP-1505: Remove unused setter.
